module/VuFind/src/VuFind/AjaxHandler/GetSearchResults.php

```diff
@@ -0,0 +1,455 @@
+     * @param ?UserRow         $user            Logged-in user
+     * @param string           $sessionId       Session ID
+     * @param SearchNormalizer $normalizer      Search normalizer
+     * @param SearchTable      $searchTable     Search table
+     * @param array            $config          Main configuration
+     * @param VersionsHelper   $versionsHelper  Versions helper
+     */
+    public function __construct(
+        SessionSettings $sessionSettings,
+        ResultsManager $resultsManager,
+        PhpRenderer $renderer,
+        RecordLoader $recordLoader,
+        ?UserRow $user,
+        string $sessionId,
+        SearchNormalizer $normalizer,
+        Search $searchTable,
+        array $config,
+        VersionsHelper $versionsHelper
+    ) {
+        $this->sessionSettings = $sessionSettings;
+        $this->resultsManager = $resultsManager;
+        $this->renderer = $renderer;
+        $this->recordLoader = $recordLoader;
+        $this->user = $user;
+        $this->sessionId = $sessionId;
+        $this->searchNormalizer = $normalizer;
+        $this->searchTable = $searchTable;
+        $this->config = $config;
+        $this->versionsHelper = $versionsHelper;
+    }
+
+    /**
+     * Handle a request.
+     *
+     * @param Params $params Parameter helper from controller
+     *
+     * @return array [response data, HTTP status code]
+     */
+    public function handleRequest(Params $params)
+    {
+        $results = $this->getSearchResults($params);
+        if (!$results) {
+            return $this->formatResponse(['error' => 'Invalid request'], 400);
+        }
+        $elements = $this->getElements($params, $results);
+        return $this->formatResponse(compact('elements'));
+    }
+
+    /**
+     * Get search results
+     *
+     * @param Params $params Request params
+     *
+     * @return ?Results
+     */
+    protected function getSearchResults(Params $params): ?Results
+    {
+        parse_str($params->fromQuery('querystring', ''), $searchParams);
+        $backend = $params->fromQuery('source', DEFAULT_SEARCH_BACKEND);
+        $searchType = $params->fromQuery('searchType', '');
+        if ('versions' === $searchType) {
+            $driverAndKeys = $this->versionsHelper->getDriverAndWorkKeysFromParams(
+                $searchParams,
+                $backend
+            );
+            if (empty($driverAndKeys['keys'])) {
+                return null;
+            }
+
+            $searchParams['lookfor'] = $this->versionsHelper
+                ->getSearchStringFromWorkKeys((array)$driverAndKeys['keys']);
+            $searchParams['type'] = $this->versionsHelper->getWorkKeysSearchType();
+        }
+
+        $results = $this->resultsManager->get($backend);
+        $paramsObj = $results->getParams();
+        $paramsObj->getOptions()->spellcheckEnabled(false);
+        $paramsObj->initFromRequest(new Parameters($searchParams));
+
+        if ('versions' !== $searchType) {
```

Comment:
Maybe use a class property, like `if (!in_array($searchType, $this->searchTypesWithoutHistory)) {` to make this more generic as well?

Reply:
Eliminated as above.